src/core/meshcop/border_agent.hpp

```diff
@@ -387,13 +410,116 @@ class BorderAgent : public InstanceLocator, private NonCopyable
 
     static Coap::Message::Code CoapCodeFromError(Error aError);
 
+    class MeshCopTxtEncoder : public InstanceLocator
```

Comment:
Move it before the methods? After `class CoapDtlsSession`? This way all nested `class` are toghther.